firestore/watch.go

```diff
@@ -446,9 +446,12 @@ func (s *watchStream) stop() {
 		return
 	}
 	if err != nil {
+		// if an error occurs while closing the stream
 		s.err = err
+	} else {
```

Comment:
In general, prefer not including else blocks: https://golang.org/doc/effective_go.html#if

Reply:
Happy to make this a return. I am going to wait for @jba to maybe weigh in. Since this is either code that should have been removed, or this branch was missed :)